### PR TITLE
[Monkey Business] Lots of Monkey Clothes-related stuff. Monkeys can also now wear glasses.

### PR DESCRIPTION
**Pull request by DeityLink**
update
http://i.4cdn.org/vg/1429319050345.webm
- Improved tons of monkey hat sprites
- Monkeys can now wear glasses (I spent quite some time making their sprites fit them, sadly they can't yet activate HUDs)
- Added a Medical Doctor Monkey Uniform
- Added a Monkey-Sized Space Suit (can currently only be spawned by admins)
- Monkeys wearing this space suit and a space-worthy hat can survive in space (they also need a mask and some oxygen of course)
- Fixed a bug that sometimes caused monkey in-hand sprites to not update when they drop an item.

I also absolute-pathed monkey/Life.dm

### CODE REVIEW

**Comment by Midaychi** [comment]
What about this stuff though? Seems like you might have been able to use this sorta code instead of having to resprite everything? https://github.com/tgstation/-tg-station/pull/8554

**Comment by DeityLink** [comment]
Some hats don't fit like others. You can't just move every sprites by a set direction and number of pixels

**Comment by ComicIronic** [comment]
I'm not a fan of making copies of clothes for monkeys. You should really be using a spritesheet system on regular clothes to make the proper icons.

And I don't see an issue with moving hats. MoMMIs do it fine.

**Comment by DeityLink** [comment]
Most hats look hideous on MoMMis (depending on their style too), that's why you always see them wearing the same hats. Also that's how it's already been for month and isn't the point of this PR. I removed tons of useless "sideway" sprites in this PR so this should even decrease the size of the rsc.

**Comment by ComicIronic** [comment]
>Also that's how it's already been for months and isn't the point of this PR.

Adding to a bad system should really be prevented, especially since it makes it harder to change as time goes on.

**Comment by RemieRichards** [comment]
@DeityLink >Some hats don't fit like others. You can't just move every sprites by a set direction and number of pixels

Except I did though, as has already been posted.
TG's monkeys can wear every single hat that a human can, with no copy pasted sprites.
The ONLY alterations I had to make to sprites were to move the Monkey's side sprites one pixel along the direction they faced (Eg: East facing moved one more pixel to the East), After I did this there were no problems with any hats.

**Comment by DeityLink** [comment]
It's the only system. I'm telling that some hats look perfect on monkeys, other hats should be lowered by 1 pixel, others by 2 pixels, some look perfect as-is when facing east or west, others need to be moved 1-2 pixels in one direction or the other.
Monkey heads aren't human heads in a different spot, their eyes are spaced differently (especially when facing east-west)

Here, to illustrate:
qed

**Comment by RemieRichards** [comment]
You dont need specific sprites for every hat in game.
If you want to bring up that point of "Oh it's slightly weird looking on the monkey" YES, because it's a monkey wearing a hat not meant for monkeys! It's cute, it makes sense (For reference check out a TG fedora monkey)

It's "Slightly better looking" Vs. Clean and extensible.

**Comment by ComicIronic** [comment]
> It's the only system. I'm telling that some hats look perfect on monkeys, other hats should be lowered by 1 pixel, others by 2 pixels, some look perfect as-is when facing east or west, others need to be moved 1-2 pixels in one direction or the other.

Then you create a general system for custom displacement. You don't resprite every hat.

**Comment by DeityLink** [comment]
@ComicIronic The cult armor rune summons "monkey-sized" cult armor and hood, so it makes sense for the hood to fit the monkey's head. you can't just displace it. And this applies for many other hats that I and other coders planned on respriting. 

Is it this big an issue to have the rsc be 167KB heavier? If yes I'll just close this PR.

**Comment by ComicIronic** [comment]
> you can't just displace it.

I disagree

And apart from that, you should include the option for a general system even if you insist on adding snowflake sprites for monkeys. Some of the current hats work fine, and there's no sense in having to copy everything over.

**Comment by DeityLink** [comment]
well you only did a displacement of 2pixels to the left so this furthers my point that you can't apply the same to every hat. Furthermore how does it even hold? 
howdoesitevenhold

>Some of the current hats work fine, and there's no sense in having to copy everything over.
For crying out loud, that's what we're already doing! And have been doing since last Halloween and no one fucking complained back then. Neither did anyone complain about it until now. And this PR only adds a fistfull of new sprites in it.
Feature freeze is in two days, can we talk about optimizing all that by then? As I said this really isn't the point of this PR. I'm sure we can come up with a solution that'll satisfy both parties (like having a clothing variables that indicates whether the sprite should be a translated version of head.dmi or if it has a custom version in monkey_head.dmi) but look, this and that, are two different matters.

This PR let's monkeys wear glasses and add a couple new monkey suits, if you want to optimize the hat sprites on monkeys' system then we can work on that without holding this PR back.

**Comment by ComicIronic** [comment]
But hats aren't the only issue.

I'm not happy with the way monkey clothes work currently, since they're all snowflake items that are just copies of items already in the game, and could be better done using a spritesheet.
The way monkey equipping in general works could be handled a lot better.

**Comment by DeityLink** [comment]
You're just talking about the hats and glasses right? Because the "uniforms" /monkeyclothes aren't copies of any items. Some of them are inspired by existing items (monkey-sized cult robes, monkey-sized space suit) but others are completely new (punpun's waiter suit)

File: code/modules/mob/living/carbon/monkey/life.dm [hunk 1]
```diff
@@ -89,620 +90,627 @@
...
+		if(M.virus2.len)
+			for (var/ID in M.virus2)
+				var/datum/disease2/disease/V = M.virus2[ID]
+				if (infect_virus2(src,V, notes="(Airborne from mucus)"))
+					return 1
+	return 0
+
+/mob/living/carbon/monkey/proc/handle_virus_updates()
+	if(status_flags & GODMODE)	return 0	//godmode
+	if(bodytemperature > 406)
```
Comment (ComicIronic) [inline code/modules/mob/living/carbon/monkey/life.dm]: You should really define this, magic numbers are a bad idea.

**Comment by ComicIronic** [comment]
>Some of them are inspired by existing items

How is this not copying?

File: code/modules/mob/living/carbon/monkey/life.dm (hunk 1 above)
Comment (DeityLink) [inline code/modules/mob/living/carbon/monkey/life.dm]: As said in the OP I absolute pathed Life.dm. I'm not the one who originally wrote that

**Comment by DeityLink** [comment]
Dude, for real? Like, What? Do you even know what copying is? The only line of code I've copied from /obj/item/clothing/suit/cultrobes to /obj/item/clothing/monkeyclothes/cultrobes is the one with the armor values. Beside that they're entirely different items. I don't see what the hell are you talking about.
And I made the monkey space suit similar to the base human space suit because I thought it'd look cute to have a human and a monkey in matching space suits, but you're not actually complaining about that are you?

**Comment by ComicIronic** [comment]
>they're entirely different items

Except they're not. If they whole point of having monkey-clothes is to allow monkeys to wear clothing, then the monkey cult robe is a copy of the regular cult robe in the sense that it is the same item, only for a different mob.

Which is, in my view, a bad way of handling clothing.
You should just have a single version of the cult robe, and designate the fact that it can be worn by monkeys.

**Comment by DeityLink** [comment]
I'm at a loss for words, It's NOT a matter of race it's a matter of SIZE! Can't you just live with the fact that there's a type of clothing that doesn't fit in any of the normal humans' slot? because the players sure can! Think of them as baby clothes, can you wear baby clothes? no you can't, because you don't fit in them! Even in dwarf fortress there are "small" versions of many clothes that can only be worn by dwarves and goblins but not by elves and humans, that's the kind of clothing that /monkeyclothes are! We added monkeyclothes as a children of clothing last october and now is not the time to put that in question.

>If they whole point of having monkey-clothes is to allow monkeys to wear clothing
It's not. It's to have monkeys wear a few specific set of clothing, not every fucking clothing in the game.

>You should just have a single version of the cult robe, and designate the fact that it can be worn by monkeys.
but, but that sounds even more snowflakky than giving monkeyclothes their own family!

**Comment by ComicIronic** [comment]
>but, but that sounds even more snowflakky than giving monkeyclothes their own family!

It wouldn't be.

>Can't you just live with the fact that there's a type of clothing that doesn't fit in any of the normal humans' slot?

Then make them children of the clothing they're meant to be copies of, slap the icons on them, and designate them as monkey clothes if you have to have them as separate items.
But having a snowflake class full of clothing that's just a version of stuff already coded into the game, but monkified, is bad because it doesn't preserve typechecking and it adds redundancy.

**Comment by DeityLink** [comment]
I'll say it again then, monkey sized cult armor is an exception, most monkey clothes aren't supposed to be based on human clothes. plus they also include their hands and feet (the doctor monkey clothing features gloves for hands and feet)

**Comment by ComicIronic** [comment]
>most monkey clothes aren't supposed to be based on human clothes

In the current code, 3/4 of the monkeyclothes are based on human versions. With your addition, it will be 2/3s. That's still more than half.

And if you want to make custom monkey items, just define new items under clothing that have a monkey-wearing flag. This is not a good argument against making a more sensible general system.

**Comment by DeityLink** [comment]
except these monkey clothes are neither uniforms nor suits, they don't fit under any previous categories, a monkey cloth can contain any disguise, it can include boots, gloves, contact lens, and many more.

Our system is already sensible enough.

**Comment by ComicIronic** [comment]
>a monkey cloth can contain any disguise, it can include boots, gloves, contact lens, and many more.

I understand that, and you'd be justified in making these special items monkey-only.

But other items ARE very similar to clothing already in the game. Something like the red monkey jumpsuit should come under the regular red jumpsuit, since it preserves pathing and reduces redundancy in the codebase.

**Comment by DeityLink** [comment]
ok...Comic...Look, if I outright delete those fucking green and red monkey jumpsuits from the game will you stop?

**Comment by d3athrow** [comment]
I'm going to go ahead and pull this with the later intention of refactoring how monkeys handle equipping clothing. Probably putting it at /carbon since aliens were at one time meant to be able to equip some things as well, even if its not re-added it would make handling their 'pockets' not snowflake.

**Comment by DeityLink** [comment]
Feel free to go at it. The ideal would be to completely reflector the way monkeys handle clothing, instead of having two completely different systems in parallel. Thanks Comic.

**Pull request merged**